lib/go-tc/deliveryservice_servers.go

```diff
@@ -114,3 +100,13 @@ type DSServer struct {
 	ServerCapabilities          []string             `json:"-" db:"server_capabilities"`
 	DeliveryServiceCapabilities []string             `json:"-" db:"deliveryservice_capabilities"`
 }
+
+type DSServerV11 struct {
+	DSServer
+	LegacyInterfaceDetails
+}
+
+type DSServerV30 struct {
```

Comment:
Why not make DSServer point to the current definition of a Delivery Service Server? Coupling to the API version means that things stop making sense unless we update that every time the API version changes; like if we have DSServerV30 then APIv3.1 comes out and no changes are made to the structure, we're either using a confusingly version structure, or we have to remember to do something like `type DSServer31 = DSServer30` and then do a find/replace to update things.